c_regex/regex.c

```diff
@@ -49,7 +49,7 @@ char **read_lines(char *filepath, int *size) {
     char buf[LINE_BUF_SIZE];
     FILE *f;
     f = fopen(filepath, "r");
-    check(f != NULL, "file open error: %s", filepath);
+    check(f != NULL, "File open error: %s", filepath);
     int cap = 10;
     char **lines = (char **)malloc(sizeof(char *) * (cap));
     check_mem(lines);
```

Comment:
should init to NULL here

    for (int i = 0; i < cap; i++)
        lines[i] = NULL;